server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -556,10 +555,35 @@ public String executor() {
     }
 
     private ExtensionsSettings readFromExtensionsYml(Path filePath) throws IOException {
-        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
-        InputStream input = Files.newInputStream(filePath);
-        ExtensionsSettings extensionSettings = objectMapper.readValue(input, ExtensionsSettings.class);
-        return extensionSettings;
+        Yaml yaml = new Yaml();
+        InputStream inputStream = Files.newInputStream(filePath);
```

Comment:
Please use try with resources. This may produce file handle leaks.

Reply:
Should the try block include just those lines or more?  The only place the method is called currently has the method in a try block

Reply:
No like that:

```java
    try (InputStream inputStream = Files.newInputStream(filePath)) {
        Map<String, Object> obj = yaml.load(inputStream);
        if (obj == null) {
            throw new IOException("extensions.yml is empty");
        }
        List<HashMap<String, ?>> unreadExtensions = new ArrayList<>((Collection<HashMap<String, ?>>) obj.get("extensions"));
        List<Extension> readExtensions = new ArrayList<Extension>();
        for (Map<String, ?> extensionMap : unreadExtensions) {
            readExtensions.add(
                new Extension(
                    extensionMap.get("name").toString(),
                    extensionMap.get("uniqueId").toString(),
                    extensionMap.get("hostName").toString(),
                    extensionMap.get("hostAddress").toString(),
                    extensionMap.get("port").toString(),
                    extensionMap.get("version").toString(),
                    extensionMap.get("description").toString(),
                    extensionMap.get("opensearchVersion").toString(),
                    extensionMap.get("javaVersion").toString(),
                    extensionMap.get("className").toString(),
                    extensionMap.get("customFolderName").toString(),
                    extensionMap.get("hasNativeController").toString()
                )
            );
        }
    }
```

Reply:
FYI I was talking about try-with-resources that ensures that files and other closeable resources are correctly closed in all error conditions.

Reply:
Oh that makes more sense